Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls.cpp

```diff
@@ -256,36 +257,71 @@ uint64_t ExecveHandler(const char* pathname, char* const* argv, char* const* env
     return -ENOEXEC;
   }
 
+  fextl::vector<const char*> EnvpArgs {};
+  char* const* EnvpPtr = envp;
+  bool AlreadyCopiedEnvp {};
+
+  auto AppendEnvpVariables = [&EnvpArgs, &AlreadyCopiedEnvp, envp, &EnvpPtr]() {
```

Comment:
This lambda is unnecessary since it's called only once.

More generally, lambda functions are a good fit for short snippets of code with few side effects. Anything more complicated would be better solved with a non-local refactor (to a global function or class) instead.

Reply:
Indeed, the visibility in to what is going to change in the future is hard here. Future PRs (seccomp and FEXServer FD inheritance) are going to call this up to three times.

Reply:
I removed the unnecessary lambda usage.